### PR TITLE
test(seed): Add Test for Unique Challenge Titles

### LINKED ISSUE #16906: [Beta] Coding Interview Algorithm Questions section - Something went wrong
#### Challenge Name

https://beta.freecodecamp.org/en/challenges/coding-interview-algorithm-questions/inventory-update

#### Issue Description

Loading any challenge from the Coding Interview Algorithm Questions section gives Something went wrong message and checking the browser console shows:
`Failed to load resource: the server responded with a status of 500 ()`

```
Error: {"output":{"message":"Cannot read property 'isComingSoon' of undefined"},"meta":{}}
    at Object.callback (http.client.js:199)
    at n (index.js:62)
    at XMLHttpRequest.i (index.js:129)
    at XMLHttpRequest.o [as __zone_symbol___onload] (zone.js:1064)
    at e.invokeTask (zone.js:424)
    at t.runTask (zone.js:191)
    at XMLHttpRequest.invoke (zone.js:486)
```

#### Browser Information

* Browser Name, Version: Google Chrome 64.0.3282.186 (Official Build) (32-bit)
* Operating System: windows 7 32-bit
* Mobile, Desktop, or Tablet: Desktop

#### Your Code

```js

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes ####

#### Description

Due to the recent problem with two challenges having the same title, I
added a check for unique titles to the test suite so that hopefully
won't happen again.

Addresses a comment in issue #16906, which was recently closed.

BREAKING CHANGE: None

### CODE REVIEW

**Comment by mstellaluna** [comment]
I quickly tested this and they are now loading with no errors. I tried the 'Find the Symmetric Difference' challenge and it works fine.

**Comment by QuincyLarson** [comment]
@scissorsneedfoodtoo Awesome! Thanks for writing a test for this! 

@Bouncey could you QA this when you get a chance?

**Review by Bouncey — APPROVED** [review]

File: seed/challengeTitles.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    this.knownTitles = [];
+  }
+  check(title) {
+    if (typeof title !== 'string') {
+      throw new Error(`Expected a valid string for ${title}, got ${typeof title}`);
+    } else if (title.length === 0) {
+      throw new Error(`Expected a title length greater than 0`);
+    }
+    const titleToCheck = title.toLowerCase().replace(/\s+/g, '');
+    const titleIndex = _.findIndex(this.knownTitles, existing => titleToCheck === existing);
```
[the remark this answers is not in the document]
Comment (scissorsneedfoodtoo) [reply seed/challengeTitles.js]: @Bouncey, that makes a lot more sense. Even though this PR is merged I'd like to make these changes. Should I open a separate PR? Or keep this branch open and force push the changes?

**Comment by scissorsneedfoodtoo** [comment]
@QuincyLarson, my pleasure! Glad I could help out a bit.

@Bouncey, thank you for the merge! However, I saw your suggestion above and would like to incorporate those changes. Let me know what you think whenever you can.

**Pull request merged**